verifiable-db/src/row_tree/api.rs

```diff
@@ -180,25 +181,48 @@ pub enum CircuitInput {
 }
 
 impl CircuitInput {
-    pub fn leaf(identifier: u64, value: U256, cells_proof: Vec<u8>) -> Result<Self> {
-        Self::leaf_multiplier(identifier, value, false, cells_proof)
+    pub fn leaf(
+        identifier: u64,
+        value: U256,
+        mpt_metadata: HashOut<F>,
+        row_unique_data: HashOut<F>,
```

Comment:
I think that to make the interface of these methods completely agnostic with respect to the source of the data, we should maybe introduce a trait like this:

```rust
pub trait RowDigestInfo {
      type RowUniqueData: Clone + ToFields;
      /// Compute the metadata associated to the column with identifier `identifier`
      fn metadata(&self, identifier: u64) -> HashOutput;
}
```
And then replace `mpt_metadata` with a `digest_info: &RowDigestInfo` and make `row_unique_data: RowDigestInfo::RowUniqueData`. 
This trait could be implemented by specific structs for each type of MPT tables we support for now (e.g., simple variables, simple mappings, mappings of mappings), but it should be generic enough to be implemented in case we will add tree construction from non-blockchain source. Wdyt? Tagging also @nikkolasg to discuss about this